### ISSUE TITLE
I can't set params `optlevel` to `1` with torch_neuronx.trace

### ISSUE THREAD

**Issue by Suprhimp** [aws-neuron/aws-neuron-samples#67]
My environment is  aws server `inf2.8xlarge`

python : 3.8.10
torch-neuronx : 2.1.1.2.0.1b0
neuronx-cc : 2.12.68.0+4480452af
 
 I'm trying to compile esrgan torch model to neuron but I have an issue.

```
from PIL import Image
import requests

import torch
import torch_neuronx
from torchvision import models
from torchvision.transforms import functional

from modules.esrgan_upscale import upscale_model_loader
import os
os.environ["NEURON_CC_FLAGS"] = "-O1"
# load the model
model = upscale_model_loader('modules/weight/4x-Ultrasharp.pth')
model.eval()

# Get an example input
image = Image.open('/home/ubuntu/diffusers-ultimate-upscale/testIm.png')
image = image.convert('RGB')
image = functional.to_tensor(image)
image = torch.unsqueeze(image, 0)

# Run inference on CPU
output_cpu = model(image)

# Compile the model
model_neuron = torch_neuronx.trace(model, image,compiler_args=['--optlevel','1'])

# Save the TorchScript for inference deployment
filename = 'model.pt'
torch.jit.save(model_neuron, filename)
```

when I run this code 
first it gives me this log

```
2024-02-20T13:36:54Z Compilation is optimized for best performance and compilation time. For faster compilation time please use -O1
```

I want to compile with `-O1` because of this error log (yes, i failed compile)
```
[XTP002] Too many instructions after unroll for function sg0000! - Compiling under --optlevel=1 may result in smaller graphs. If you are using a transformer model, try using a smaller context_length_estimate value.
```

I can't set the optlevel flag to 1 ... even I changed inside the module [code](https://github.com/gschaves/torch-neuronx/blob/e57f3e8d0cc9dabd7119326b0a450d2e6e58b843/torch_neuronx/xla_impl/trace.py#L242C5-L253C34) like this
```
    command = [
        neuron_cc,
        "compile",
        filename,
        "--framework",
        "XLA",
        "--target",
        "trn1",
        "--output",
        neff_filename,
        "--optlevel",
        "1"
    ]
    command.extend(compiler_args)
```

 what should I do if I want to compile with --optlevel=1 with `torch_neuronx.trace` ?

**Comment by aws-donkrets** [comment]
Hi [Suprhimp](https://github.com/Suprhimp), Perhaps a typo in your command assignment. The inf2/trn1 Neuron compiler command is named 'neuronx-cc'. In your command stmt I see 'neuron_cc'.

**Comment by aws-donkrets** [comment]
Hi [Suprhimp](https://github.com/Suprhimp), took a quick look at your code and it seems to be correct. 
The ``torch_neuronx.trace`` call can pass compiler options and the way you have done it looks correct as does your  command definition.  I'll note you don't need to use the ``os.environ["NEURON_CC_FLAGS"] = "-O1"``  line so that can be removed. One suggestion is to move the ``neff_filename`` parameter to the end of the ``command`` setting, allowing all the cmd-line flags to appear before the filename. So, the cmd-line would look like:
   ``neuronx_cc compile input_file_name --framework XLA --target trn1 --optlevel 1 --output neff_filename``

Another suggestion would be to run the above command by hand to see if you get the same result.

**Comment by Suprhimp** [comment]
Hi, Thanks for checking my issue  @aws-donkrets  :)



even if I change the code like this
in `trace.py` function name  `hlo_compile` like this

```
if neuron_cc is None:
        raise RuntimeError("neuronx-cc compiler binary does not exist")
    command = [
        neuron_cc,
        "compile",
        filename,
        "--framework",
        "XLA",
        "--target",
        "trn1",
        "--optlevel",
        "1",
        "--output",
        neff_filename,
    ]
    command.extend(compiler_args)
```


it gives me this log 
```
2024-02-29T02:01:25Z Compilation is optimized for best performance and compilation time. For faster compilation time please use -O1
```

and also I faild my compile ;)

**Comment by Suprhimp** [comment]
@aws-donkrets  hello, let me add question, Is there any way to compile `.pth` file to run my torch file in `inf2` instance?

faster compile flag still not work.

Can you check it please?

**Comment by aws-taylor** [comment]
Hello @Suprhimp,

 We do not directly support compiling `.pth` files, you would need to load it first, perhaps using `load_state_dict()`, then trace the loaded model to trigger compilation.

 Could you share your model or more of the failure logs from the compiler (usually log-neuronx-cc.txt)? That will give us more of an idea of why the failure is occurring.

**Comment by nalwayaakshay** [comment]
Hello @Suprhimp ,
It's been more than an year since we've heard from you. We will plan to close this issue within the next 2 weeks if we don't hear back. Thanks.

**Comment by jeffhataws** [comment]
Hi @Suprhimp ,

Looks like this model can't be traced using the trace API. Furthermore, using an input image that's too large also lead to "Too many instructions" error you see above, regardless of -O (or --optlevel) option.

I am able to compile the model using the torch-xla lazy evaluation mode and smaller image. Here's an example using https://huggingface.co/lokCX/4x-Ultrasharp/blob/main/4x-UltraSharp.pth model and ModelLoader from spandrel (pip install spandrel):

```
from PIL import Image
import requests

import torch
import torch_neuronx
import torch_xla.core.xla_model as xm
from torchvision import models
from torchvision.transforms import functional

from spandrel import ModelLoader

import os
os.environ["NEURON_CC_FLAGS"] = "--model-type=unet-inference"
# load the model
model = ModelLoader().load_from_file('./4x-UltraSharp.pth')
model.eval()
model.to("xla")

# Get an example input
image = Image.open('./simple_sky-blue_02-512x512.png')
image = image.convert('RGB')
image = functional.to_tensor(image)
image = torch.unsqueeze(image, 0)
image = image.to("xla")

# Run inference on CPU
output = model(image)
xm.mark_step()
print(output.shape)

```

(Example image is https://opengameart.org/sites/default/files/oga-textures/146572/simple_sky-blue_02-512x512.png)

I haven't try larger sizes. Since this is an upscaling application, please try to see if this works for you. We will close this ticket for now. Let us know if we can be of further help.
